### PR TITLE
feat: add merge on green to gatsbyjs/gatsby repo

### PR DESCRIPTION
**Pull request by pieh**
fixes https://github.com/gatsbyjs/gatsby/issues/13022

### CODE REVIEW

File: rules/merge-on-green.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { danger, peril } from "danger"
+import * as octokit from "@octokit/rest"
+
+const checkPRCondidationsAndMerge = async ({
```
Comment (sidharthachatterjee) [inline rules/merge-on-green.ts]: What is condidations 😛

File: rules/merge-on-green.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }
+}
+
+export const mergeOnGreen = async () => {
+  try {
+    if (danger.github.action === `completed` && danger.github.check_suite) {
+      // this is for check_suite.completed
+
+      // search returns first 100 results, we are not handling pagination right now
+      // because it's unlikely to get more 100 results for given sha
```
Comment (sidharthachatterjee) [inline rules/merge-on-green.ts]: For future reference: 

We initially tried using the `status` event but it wasn't being sent as a web hook into peril. We therefore find the pull request including this commit SHA.

File: rules/merge-on-green.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        q: `${danger.github.check_suite.head_sha} is:open repo:${
+          danger.github.repository.owner.login
+        }/${danger.github.repository.name}`,
+      })
+
+      let i = 0
+      while (i < results.data.items.length) {
+        const pr = results.data.items[i]
+        i++
+        await checkPRCondidationsAndMerge({
```
Comment (sidharthachatterjee) [inline rules/merge-on-green.ts]: ☕️

File: rules/merge-on-green.ts (hunk 2 above)
Comment (orta) [reply rules/merge-on-green.ts]: I'd recommend switching this out:

```
    const userAuthedAPI = new octokit()
	    userAuthedAPI.authenticate({
	      type: "token",
	      token: peril.env.GITHUB_ACCESS_TOKEN,
	    })
	    await userAuthedAPI.pullRequests.merge({
	      merge_method: `squash`,
	      commit_title: `${pr.data.title} (#${number})`,
	      number,
	      owner,
	      repo,
	    })
	  }
	```

and making it just 

```
danger.github.api.pullRequests.merge
```

and then you can remove that access token

**Pull request merged**